### PR TITLE
`deno doc` parses super-class names

### PR DESCRIPTION
**Pull request by ondras**
Well.

This is my noobish attempt at making `deno doc` parse the `extends` clause of a class definition. Incidentally, it is also my very first rust code ever written. Still no idea how the Box stuff is supposed to function.

It seems to work and I also added a test for the feature.

### CODE REVIEW

File: cli/doc/printer.rs [hunk 1]
```diff
@@ -427,11 +427,23 @@ fn format_function_signature(node: &doc::DocNode, indent: i64) -> String {
...
+  let class_def = node.class_def.clone().unwrap();
+  let super_suffix = if class_def.super_class.is_some() {
+    format!(
+      " {} {}",
+      colors::magenta("extends".to_string()),
+      colors::bold(class_def.super_class.unwrap())
+    )
+  } else {
+    String::from("")
+  };
```
Comment (lucacasonato) [inline cli/doc/printer.rs:439]: `if let` would be good for this:

```suggestion
  let super_suffix = if let Some(super_class) = class_def.super_class {
    format!(
      " {} {}",
      colors::magenta("extends".to_string()),
      super_class
    )
  } else {
    String::from("")
  };
```

**Comment by ondras** [comment]
> Looks good :-D

Great, thanks ;)

> Could you add a check to the bottom of that test to test that the output of the printer includes `class Foobar extends Fizz`?

Yes, will add.

**Comment by ondras** [comment]
Test added.

**Comment by ondras** [comment]
Great! @bartlomieju , shall I continue with exposing `implements` as well?

**Comment by bartlomieju** [comment]
> Great! @bartlomieju , shall I continue with exposing `implements` as well?

@ondras sure

**Comment by ondras** [comment]
> > Great! @bartlomieju , shall I continue with exposing `implements` as well?
> 
> @ondras sure

Continuing in https://github.com/denoland/deno/pull/4604

**Pull request merged**